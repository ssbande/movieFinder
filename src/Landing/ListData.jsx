import React from 'react';
import PropTypes from 'prop-types';
import { Grid } from 'semantic-ui-react';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';

import NoData from './NoData';
import DataItem from './DataItem';
import { HistoryProps } from '../utils/propTypes';

const ListData = props => {
  const { activeItem, history } = props;
  const selectedData = props[activeItem] || [];

  const handleCardClick = id => history.push(`${activeItem}/${id}`);

  return <Grid stackable columns={4}>
    {!selectedData.length
      ? <NoData />
      : selectedData.map(item => (
        <DataItem key={item.id} item={item} handleCardClick={handleCardClick} />
      ))
    }
  </Grid>
}

ListData.propTypes = {
  activeItem: PropTypes.string.isRequired,
  history: HistoryProps,
  movies: PropTypes.array,
  shows: PropTypes.array,
  artists: PropTypes.array
}

export default withRouter(connect()(ListData));