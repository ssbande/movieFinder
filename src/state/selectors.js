import get from 'lodash/get';
import constants from '../utils/constants';
const { config } = constants;

export const getQuery = state => get(state, 'userQuery', null);
export const getFilter = state => get(state, 'filter', null);
export const getSubmitted = state => get(state, 'submitted', false);
export const getSuggestions = state => get(state, 'suggestions', []);
export const getMovies = state => get(state, 'movies.results', []);
export const getShows = state => get(state, 'shows.results', []);
export const getArtists = state => get(state, 'artists.results', []);
export const getDetailInfo = state => get(state, 'detailInfo', null);
export const getError = state => get(state, 'error', false);

export const getInitialMenu = state => {
  const filter = getFilter(state);
  return filter === config.all.category
    ? config.movie.route
    : config[filter].route;
}

export const getMenus = state => {
  const filter = getFilter(state);
  return Object.entries(config)
    .filter(item => {
      return (filter !== 'all') 
        ? item[0] === filter
        : item[0] !== 'all';
    });
}

export const shouldShowSuggestions = state => {
  const { queryLengthToStartSuggestions } = constants;
  const query = getQuery(state);
  const isSubmitted = getSubmitted(state);

  return query.length >= queryLengthToStartSuggestions && !isSubmitted;
}

export const shouldShowStartScreen = state => {
  const movies = getMovies(state);
  const shows = getShows(state);
  const artists = getArtists(state);
  const submitted = getSubmitted(state);

  return !movies.length && !shows.length && !artists.length && !submitted;
}