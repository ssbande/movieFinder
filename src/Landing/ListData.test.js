import React from 'react';
import { shallow } from 'enzyme';
import { makeData } from '../state/selectors.test';

import ListData from './ListData';

const setup = () => ({
  props: {
    activeItem: 'movies',
    history: { push: jest.fn() },
    movies: makeData('movie'),
    shows: makeData('show'),
    artists: makeData('artist'),
  }
})

describe('List Data tests', () => {
  it('should render correctly for movie view', () => {
    const { props } = setup();
    const wrapper = shallow(<ListData {...props} />);
    expect(wrapper).toMatchSnapshot();
  })

  it('should render correctly for TV shows view', () => {
    const { props } = setup();
    props.activeItem = 'shows';
    const wrapper = shallow(<ListData {...props} />);
    expect(wrapper).toMatchSnapshot();
  })

  it('should render correctly for person view', () => {
    const { props } = setup();
    props.activeItem = 'artists';
    const wrapper = shallow(<ListData {...props} />);
    expect(wrapper).toMatchSnapshot();
  })

  it('should render correctly for no data view', () => {
    const { props } = setup();
    props.movies = [];
    const wrapper = shallow(<ListData {...props} />);
    expect(wrapper).toMatchSnapshot();
  })
})