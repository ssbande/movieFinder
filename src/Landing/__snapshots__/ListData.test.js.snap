// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List Data tests should render correctly for TV shows view 1`] = `ShallowWrapper {}`;

exports[`List Data tests should render correctly for movie view 1`] = `ShallowWrapper {}`;

exports[`List Data tests should render correctly for no data view 1`] = `ShallowWrapper {}`;

exports[`List Data tests should render correctly for person view 1`] = `ShallowWrapper {}`;
